graphdatascience/graph_data_science.py

```diff
@@ -92,6 +93,13 @@ def __init__(
         try:
             server_version_string = self._query_runner.run_query("RETURN gds.version()", custom_error=False).squeeze()
         except Exception as e:
+            if "Unknown function 'gds.version'" in str(e):
+                raise GdsNotFound(
```

Comment:
A further improvement could be to check if the license status is correct on `__init__`